chips/nrf52/src/uart.rs

```diff
@@ -401,19 +403,18 @@ impl kernel::hil::uart::UART for Uarte {
         let regs = unsafe { &*self.regs };
 
         // truncate rx_len if necessary
-        let mut length = rx_len;
-        if rx_len > rx_buf.len() {
-            length = rx_buf.len();
-        }
+        let mut truncated_length = core::cmp::min(rx_len, rx_buf.len());
+        truncated_length = core::cmp::min(truncated_length, 255);
```

Comment:
we actually need both these variables:
* truncated_length - min(buffer_len, actual len)
* truncated_uart_max_length - min(truncated_length, UART_MAX_BUFFER)

See comment below!